core/src/main/kotlin/org/evomaster/core/output/TestSuiteOrganizer.kt

```diff
@@ -13,22 +20,131 @@ import org.evomaster.core.search.Solution
  * name each test will have
  */
 class TestSuiteOrganizer {
-
     companion object {
-        fun sortTests(solution: Solution<*>): List<TestCase> {
+        val sortingHelper = SortingHelper()
+        val namingHelper = NamingHelper()
+
+
+        fun sortTests(solution: Solution<*>, customNaming: Boolean = false): List<TestCase> {
 
             //TODO here in the future we will have something bit smarter
-            return naiveSorting(solution)
+            return sortingHelper.sort(solution, namingHelper, customNaming)
+        }
+    }
+}
+
+class NamingHelper {
+    private fun criterion1_500 (individual: EvaluatedIndividual<*>): String{
+        var found500 = false
+        individual.results.forEach {
+            if((it as RestCallResult).getStatusCode() == 500) found500 = true
```

Comment:
this (ie the "as" cast) will break in the future as soon as we address something more than REST. Should rather use a filterIsInstance on the stream

Reply:
Using filterIsInstance instead. One exception, for maxStatusCodeComparator, I am using maxBy, which is nullable. As a result, a casting is needed, though it comes after a filterIsInstance, so the worst it can be is null.